driver/src/main/java/org/neo4j/driver/internal/messaging/request/RouteMessage.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2002-2020 "Neo4j,"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver.internal.messaging.request;
+
+import java.util.Map;
+import java.util.Objects;
+
+import org.neo4j.driver.Value;
+import org.neo4j.driver.internal.messaging.Message;
+
+import static java.util.Collections.unmodifiableMap;
+
+/**
+ * From the application point of view it is not interesting to know about the role a member plays in the cluster. Instead, the application needs to know which
+ * instance can provide the wanted service.
+ * <p>
+ * This message is used to fetch this routing information.
+ */
+public class RouteMessage implements Message
+{
+    public final static byte SIGNATURE = 0x66;
+    private final Map<String,Value> routingContext;
+    private final String databaseName;
+
+    /**
+     * Constructor
+     *
+     * @param routingContext The routing context used for multi-data center deployments. It should be used in combination with multi-data center load
```

Comment:
nit: possibly routing context will have other applications beyond multi-data center in the future so we could be more generic here